en/05_Contributing/05_Coding_Conventions/01_PHP_Coding_Conventions.md

```diff
@@ -6,190 +6,157 @@ iconBrand: php
+- `Task` ([`BuildTask`](api:SilverStripe\Dev\BuildTask))
 
-Use [phpdoc](https://phpdoc.org/) syntax before each definition (see [tutorial](https://manual.phpdoc.org/HTMLSmartyConverter/HandS/phpDocumentor/tutorial_phpDocumentor.quickstart.pkg.html)
-and [tag overview](https://manual.phpdoc.org/HTMLSmartyConverter/HandS/phpDocumentor/tutorial_tags.pkg.html)).
+## Directory naming - src dir
 
- * Methods should include at least `@param` and `@return`.
- * Include a blank line after the description.
- * Use `{@link MyOtherClass}` and `{@link MyOtherClass->otherMethod}` for inline references.
- * Denote preformatted code examples in `<code></code>` blocks.
- * Always start block-level comments containing phpdoc with two asterisks (`/** ... */`).
+- Follow the [PSR-4 coding standard](https://www.php-fig.org/psr/psr-4/)
+- Use initial caps pluralised for the directory naming. For example, use `Controllers` rather than `Controller` or `controllers`
+- Put similar types of classes in the same directory. For example, put Controllers in a `Controllers` directory
+- Put DataObject subclasses in a directory called `Models`
+- If you end up with a lot of classes of a specific type with some clear grouping then you can use subdirectories to group them. For example use `src/Models/Sports` if you have a lot of sports related models.
 
-Example:
+## Properties and accessors
 
+- For setter methods use a combination of a `static` return type and `return $this` to allow for fluent programming.
+- Do not use [dynamic properties](https://www.php.net/manual/en/language.oop5.properties.php#language.oop5.properties.dynamic-properties)
 
-```php
-/**
- * My short description for this class.
- * My longer description with
- * multiple lines and richer formatting.
- *
- * Usage:
- * <code>
- * $c = new MyClass();
- * $c->myMethod();
- * </code>
- *
- * @package custom
- */
-class MyClass extends Class
-{
-    /**
-     * My Method.
-     * This method returns something cool. {@link MyParentMethod} has other cool stuff in it.
-     *
-     * @param string $colour The colour of cool things that you want
-     * @return DataList A list of everything cool
-     */
-    public function myMethod($colour)
-    {
-        // ...
-    }
+## Method, property, and constant visibility
 
-}
-```
+- Avoid `public` properties, instead use a combination of `private` (or `protected`) properties and `public` getter and setter methods.
+- Prefer `private` over `protected` for all new properties. Subclasses can access the properties through the public getter and setter methods.
+- Prefer `private` over `protected` for new methods, unless:
+    - There is a clear use case for customising the specific functionality of that method on its own and the method implementation is unlikely to change in the near future
+    - The method is intended to be called by subclasses.
+- Prefer `private` over `protected` for constants, unless they are intended to be used in subclasses.
```

Comment:
Probably need to note to use public for constants that should be accessed outside the class hierarchy, otherwise it sounds like we only ever use private or protected for constants and therefore have to use other ways (such as getter methods) to access that data from external classes.

Reply:
Updated